### ISSUE TITLE
"Download all comments"

### ISSUE THREAD

**Issue by rjolina** [sameerdash2/comment-viewer#2]
It would be great if there was an option "Download all comments" that would generate all comments (including comments on comments) in the form of an HTML or TXT file.

**Comment by sameerdash2** [comment]
Interesting idea, very possible since it already stores the comments in a database. The main concern would be the amount of computational work at each stage: 1) querying the comments from database, 2) formatting that raw array into something presentable, and 3) writing that to a file. 

It'd work fine for 1000 comments, but at 100,000+ comments we'd have to introduce some kind of chunking so it doesn't freeze the app.

Including replies may not be feasible -- I don't fetch/store replies at all (unless the user clicks on it), because they take up extra storage and they're more expensive to request from the API.

**Comment by emviem** [comment]
I was using another one (youtubecommentsdownloader) that was free for some time but now it has turned to a paid service. Your free service is preferable to me and thank you for this. One small problem for me is that "replies" to comments arent shown automatically but one must click one by one to expand a reply/replies. But I just read your above explanation about it and I think it is not feasible. Nonetheless, I use commenviewer.com happily, thank you.

**Comment by emviem** [comment]
Youtube's recent limiting API change is frustrating.

**Comment by sameerdash2** [comment]
Yeah, the site is already approaching limits (both disk storage and API quota) without fetching replies automatically.

I'm not familiar with youtubecommentsdownloader, but seeing how their count of loaded comments only increments by 20 at a time, I would guess they're also affected by the 100 -> 20 API change.
